ractor.rb

```diff
@@ -1,95 +1,101 @@
+# For example, unlike in threads, ractors can't access all the objects available in other ractors. Even objects normally
+# available through variables in the outer scope are prohibited from being used across ractors.
 #
 #     a = 1
 #     r = Ractor.new {puts "I am in Ractor! a=#{a}"}
 #     # fails immediately with
 #     # ArgumentError (can not isolate a Proc because it accesses outer variables (a).)
 #
+# The object must be explicity shared:
+#     a = 1
+#     r = Ractor.new(a) { |a1| puts "I am in Ractor! a=#{a1}"}
+#
 # On CRuby (the default implementation), Global Virtual Machine Lock (GVL) is held per ractor, so
-# ractors are performed in parallel without locking each other.
+# ractors can perform in parallel without locking each other. This is unlike the situation with threads
+# on CRuby.
 #
-# Instead of accessing the shared state, the objects should be passed to and from ractors via
-# sending and receiving objects as messages.
+# Instead of accessing shared state, objects should be passed to and from ractors by
+# sending and receiving them as messages.
 #
 #     a = 1
 #     r = Ractor.new do
-#       a_in_ractor = receive # receive blocks till somebody will pass message
+#       a_in_ractor = receive # receive blocks until somebody passes a message
 #       puts "I am in Ractor! a=#{a_in_ractor}"
 #     end
 #     r.send(a)  # pass it
 #     r.take
-#     # here "I am in Ractor! a=1" would be printed
+#     # Here, "I am in Ractor! a=1" would be printed
 #
 # There are two pairs of methods for sending/receiving messages:
 #
 # * Ractor#send and Ractor.receive for when the _sender_ knows the receiver (push);
 # * Ractor.yield and Ractor#take for when the _receiver_ knows the sender (pull);
 #
-# In addition to that, an argument to Ractor.new would be passed to block and available there
-# as if received by Ractor.receive, and the last block value would be sent outside of the
+# In addition to that, any arguments passed to Ractor.new are passed to the block and available there
+# as if received by Ractor.receive, and the last block value is sent outside of the
 # ractor as if sent by Ractor.yield.
 #
-# A little demonstration on a classic ping-pong:
+# A little demonstration of a classic ping-pong:
 #
-#     server = Ractor.new do
+#     server = Ractor.new(name: "server") do
 #       puts "Server starts: #{self.inspect}"
 #       puts "Server sends: ping"
 #       Ractor.yield 'ping'                       # The server doesn't know the receiver and sends to whoever interested
 #       received = Ractor.receive                 # The server doesn't know the sender and receives from whoever sent
 #       puts "Server received: #{received}"
 #     end
 #
-#     client = Ractor.new(server) do |srv|        # The server is sent inside client, and available as srv
+#     client = Ractor.new(server, name: "client") do |srv|  # The server is sent to the client, and available as srv
```

Comment:
I think it's easy to miss (in this longer line of text) how important `|srv|` is to define what `srv` means in the following.

Reply:
Makes sense.